spec/patterns/aws_spec.rb

```diff
@@ -0,0 +1,394 @@
+      should include("x_edge_result_type" => "Miss" )
+      should include("x_edge_request_id" => "UGskZ6dUKY7b4C6Pt7wAWVsU2KO-vTRe-mR4r9H-WQMjhNvY6w1Xcg==" )
+      should include("x_host_header" => "host.example.com" )
+      should include("cs_protocol" => "https" )
+      should include("cs_bytes" => 883 )
+      should include("time_taken" => 0.036 )
+      should include("x_forwarded_for" => "-" )
+      should include("ssl_protocol" => "TLSv1.2" )
+      should include("ssl_cipher" => "ECDHE-RSA-AES128-GCM-SHA256" )
+      should include("x_edge_response_result_type" => "RefreshHit" )
+    end
+  end
+
+  ["tags", "params"].each do |attribute|
+    it "have #{attribute} as nil" do
+      expect(subject[attribute]).to be_nil
+    end
+  end
+
+  context 'version 1.0' do # more fields at the end
+
+    let(:message) do
+      # Version: 1.0  -  https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html
+      # Fields: date time x-edge-location sc-bytes c-ip cs-method cs(Host) cs-uri-stem sc-status cs(Referer) cs(User-Agent) cs-uri-query cs(Cookie) x-edge-result-type x-edge-request-id x-host-header cs-protocol cs-bytes time-taken x-forwarded-for ssl-protocol ssl-cipher x-edge-response-result-type cs-protocol-version fle-status fle-encrypted-fields c-port time-to-first-byte x-edge-detailed-result-type sc-content-type sc-content-len sc-range-start sc-range-end
+      "2019-12-04	21:02:31	LAX1	392	192.0.2.100	GET	d111111abcdef8.cloudfront.net	/index.html	200	-	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36	-	-	Hit	SOX4xwn4XV6Q4rgb7XiVGOHms_BGlTAC4KyHmureZmBNrjGdRLiNIQ==	d111111abcdef8.cloudfront.net	https	23	0.001	-	TLSv1.2	ECDHE-RSA-AES128-GCM-SHA256	Hit	HTTP/2.0	-	-	11040	0.001	Hit	text/html	78	-	-"
+    end
+
+    it 'matches' do
+      should include("timestamp" => "2019-12-04\t21:02:31")
+
+      if ecs_compatibility?
+        should include("destination"=>{"bytes"=>392}, "source"=>{"ip"=>"192.0.2.100", "bytes"=>23, "port"=>11040}) # source.port not matched in legacy mode
+        should include("url"=>{"domain"=>"d111111abcdef8.cloudfront.net", "path"=>"/index.html"})
+        should include("http"=>hash_including("request"=>{"mime_type"=>"text/html", "method"=>"GET"}, "response"=>{"status_code"=>200})) # mime_type not matched in legacy mode
+        should include("user_agent"=>{"original"=>"Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36"})
+        should include("tls"=>{"cipher"=>"ECDHE-RSA-AES128-GCM-SHA256"})
+        should include("network"=>{"protocol"=>"https"})
+
+        should include("http"=>hash_including("version"=>"2.0"))
+        should include("aws"=>{"cloudfront"=>{
+            "x_edge_location"=>"LAX1",
+            "x_edge_request_id"=>"SOX4xwn4XV6Q4rgb7XiVGOHms_BGlTAC4KyHmureZmBNrjGdRLiNIQ==", # event.id
```

Comment:
personally, do not like keeping the `x_edge_` prefix esp. for the `request_id` as it's a unique identifier (could be used as `event.id`) and sounds a bit unclear with the prefix, believe it's fine to strip the prefix for these 4 fields. any preferences?

Reply:
With these fields being nested as custom fields under `aws.cloudfront.*`, I could see an argument for keeping the `x-edge-*` prefix naming aligning with how AWS defines the fields in CloudFront's log format.

Reply:
Thanks for the feedback Eric, 'outside' feedback always helps esp. with naming things.
Let's keep the x_edge_ prefix than.